src/backend/dsg_updater.cpp

```diff
@@ -134,30 +144,62 @@ void DsgUpdater::callUpdateFunctions(size_t timestamp_ns, UpdateInfo::ConstPtr i
 
     functor->call(*source_graph_, *target_dsg_, info);
     if (hooks.find_merges && enable_merging) {
+      auto& tracker = merge_tracker.getMergeGroup(name);
+
       // TODO(nathan) handle given merges
       const auto merges = hooks.find_merges(*source_graph_, info);
       const auto applied =
-          merge_tracker.applyMerges(*source_graph_, merges, *target_dsg_, hooks.merge);
+          tracker.applyMerges(*source_graph_, merges, *target_dsg_, hooks.merge);
       VLOG(1) << "[Backend: " << name << "] Found " << merges.size()
               << " merges (applied " << applied << ")";
 
       if (config.enable_exhaustive_merging) {
         size_t merge_iter = 0;
         size_t num_applied = 0;
         do {
+          if (name == "surface_places") {
+            continue;
+          }
```

Comment:
Reminder that we talked about moving this to be a flag in the update functors themselves instead of doing this here by hard-coding a functor name (that isn't tied to the actual layer the functor operates on)